src/Behat/Context/LogsContext.php

```diff
@@ -87,22 +234,80 @@ public function printWatchdogLogs(array $logs) {
    * @return \Drupal\Core\GeneratedUrl|string
    *   Generated url.
    */
-  protected function getDblogEventUrl(int $wid) {
+  protected static function getDblogEventUrl(int $wid) {
     $options = ['absolute' => TRUE];
-    if (!empty($this->baseUrl)) {
-      $options['base_url'] = $this->baseUrl;
+    if (!empty(self::$baseUrl)) {
+      $options['base_url'] = self::$baseUrl;
     }
 
-    // It is not possible to invoke core methods because the way the url generated is not compatible:
+    // It is not possible to invoke core methods because the way
+    // the url generated is not compatible:
     // - In Drupal 7 it's used the relative path.
     // - In Drupal 8 it's used the routing system.
-    if ($this->getCore() instanceof Drupal7) {
+    if (self::getDrupalVersion() == 7) {
       return url('/admin/reports/event/' . $wid, $options);
     }
     else {
-      return Url::fromRoute('dblog.event', ['event_id' => $wid], $options)->toString();
+      return Url::fromRoute('dblog.event', [
+        'event_id' => $wid,
+      ], $options)->toString();
     }
 
   }
 
+  /**
+   * Determine major Drupal version.
+   *
+   * @return int
+   *   The major Drupal version.
+   *
+   * @throws \Drupal\Driver\Exception\BootstrapException
+   *   Thrown when the Drupal version could not be determined.
+   *
+   * @see \Drupal\Driver\DrupalDriver::getDrupalVersion
+   */
+  public static function getDrupalVersion() {
```

Comment:
I understand this method is necessary when we are working on static contexts, as we publish the report after the Behat suite.  Looks good!

Can we move this method to a new class? It may be needed by more Contexts than the LogsContext. Maybe not now, but in the future, and is a quick change.

Reply:
I moved it into the RawDrupalContext;